service-base/src/main/java/org/eclipse/hono/service/AbstractServiceBase.java

```diff
@@ -65,6 +67,17 @@ public final void setTracer(final Tracer opentracingTracer) {
         this.tracer = Objects.requireNonNull(opentracingTracer);
     }
 
+    /**
+     * Sets the HealthCheckServer to be used.
+     * 
+     * @param healthCheckServer The HealthCheckServer.
+     * @throws NullPointerException if healthCheckServer is {@code null}.
+     */
+    @Autowired
```

Comment:
I wonder if this could/should be marked `required = false`. If no health check server is registered, the service will still work properly, wouldn't it?